feat: adds secp256k1 keypair type to ipfs key gen command

imthe-1 opened the pull request:
As per discussions with @2color we found out that we can not create secp256k1 IPNS keypairs as of now and since we can create k1 keypairs using js-ipfs/ipfs-core, we have add the ability here as well.

This might be helpful for projects, like ours, that want to derive their IPNS keypairs from a seed phrase.

Jorropo reviewed (changes requested):
Thx, I think this deserves a test in https://github.com/ipfs/kubo/blob/master/test/sharness/t0027-rotate.sh and https://github.com/ipfs/kubo/blob/master/test/sharness/t0165-keystore.sh.

imthe-1 commented:
yes @lidel. will rebase and update the PR for sure. planning to close this by tomorrow.

got it. thanks!

imthe-1 commented:
@Jorropo @lidel guys I have added the test cases, rebased and updated the PR. 

168 test cases got added in https://github.com/ipfs/kubo/blob/master/test/sharness/t0027-rotate.sh by only making some minor additions.
4 test cases added to https://github.com/ipfs/kubo/blob/master/test/sharness/t0165-keystore.sh with a bit of struggle on the key import and export test cases. and this is where I need help from you guys.

Key import and export test cases are still missing since I ran into the following error when running the export command:

```
./ipfs key export -f pem-pkcs8-cleartext key2
Error: marshalling key to PKCS8 format: x509: unknown key type while marshaling PKCS#8: *crypto.Secp256k1PrivateKey
```

seems like the crypto/x509 library couldn't export secp256k1 private key in PKCS8 format. Please have a look and let me know if any further changes are required from my side🙏

Also, I am curious why do we have a openssl_secp384r1.pem file here(https://github.com/ipfs/kubo/blob/master/test/sharness/t0165-keystore-data/openssl_secp384r1.pem) and is the info mentioned in the readme(https://github.com/ipfs/kubo/blob/master/test/sharness/t0165-keystore-data/README.md) still valid?

The t0165-keystore.sh file treats openssl_secp384r1.pem as an INVALID KEY. can you please help me understand its purpose please.

imthe-1 commented:
@Jorropo @lidel guys are we good to merge this?

Jorropo reviewed (approved):
Looks fine to me, I havn't checked this is actually working myself but I belive I'll hear about it if it doesn't.

Thx for your patch :heart: :tada:

Jorropo commented:
@BigLep it's a minor QOL, Ed25519 is still our recommended key type I don't want to push anyone to start using secp256k1 just because it's on a changelog.
On the other hand if you wanted to use your bitcoin address as your IPFS peerid as your IPNS name, I guess you would want to know about it.

So yes, with a test saying you now have the option ?
